Remove duplicate DNS lookup implementation

achristensen07 opened the pull request:
#### 5a0dc0c7e3a5b1d90ad25e7901711012c173e4e4
<pre>
Remove duplicate DNS lookup implementation
<a href="https://bugs.webkit.org/show_bug.cgi?id=257303">https://bugs.webkit.org/show_bug.cgi?id=257303</a>
rdar://109812699

Reviewed by Youenn Fablet.

DNSServiceGetAddrInfo is the replacement for CFHostStartInfoResolution.

Remove NetworkRTCResolver in favor of implementing DNSResolveQueueCFNet::resolve
like the libSoup ports and sharing 1 implementation instead of having duplicate code.

* Source/WebCore/platform/network/DNS.cpp:
(WebCore::resolveDNS):
* Source/WebCore/platform/network/DNS.h:
(WebCore::IPAddress::fromSockAddrIn6): Deleted.
* Source/WebCore/platform/network/DNSResolveQueue.h:
* Source/WebCore/platform/network/cf/DNSResolveQueueCFNet.cpp:
(WebCore::DNSResolveQueueCFNet::CompletionHandlerWrapper::create):
(WebCore::DNSResolveQueueCFNet::CompletionHandlerWrapper::complete):
(WebCore::DNSResolveQueueCFNet::CompletionHandlerWrapper::addIPAddress):
(WebCore::DNSResolveQueueCFNet::CompletionHandlerWrapper::CompletionHandlerWrapper):
(WebCore::extractIPAddress):
(WebCore::dnsLookupCallback):
(WebCore::DNSResolveQueueCFNet::performDNSLookup):
(WebCore::DNSResolveQueueCFNet::platformResolve):
(WebCore::DNSResolveQueueCFNet::resolve):
(WebCore::DNSResolveQueueCFNet::stopResolve):
(WebCore::clientCallback): Deleted.
* Source/WebCore/platform/network/cf/DNSResolveQueueCFNet.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.cpp:
(WebKit::NetworkRTCProvider::close):
(WebKit::NetworkRTCProvider::createResolver):
(WebKit::NetworkRTCProvider::stopResolver):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCResolver.cpp: Removed.
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCResolver.h: Removed.
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCResolverCocoa.cpp: Removed.
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCResolverCocoa.h: Removed.
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/264586@main">https://commits.webkit.org/264586@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1e6a4acafc500ab1207343516096545f2c17cb52

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/8033 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/8318 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/8530 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9685 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/8137 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/8037 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/10304 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/8229 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/11003 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/8177 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/9263 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/7335 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9808 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/6564 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/7355 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/14936 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/7687 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/7479 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/10841 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7945 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/6467 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/7259 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/11467 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/962 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7686 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/platform/network/cf/DNSResolveQueueCFNet.cpp:
```diff
@@ -67,38 +71,109 @@ void DNSResolveQueueCFNet::updateIsUsingProxy()
+        return std::nullopt;
+    }
+    if (address->sa_family == AF_INET)
+        return IPAddress { reinterpret_cast<const struct sockaddr_in&>(address).sin_addr };
```
```suggestion
        return IPAddress { reinterpret_cast<const struct sockaddr_in*>(address)->sin_addr };
```

youennf commented on Source/WebCore/platform/network/cf/DNSResolveQueueCFNet.cpp:
```diff
@@ -67,38 +71,109 @@ void DNSResolveQueueCFNet::updateIsUsingProxy()
+    if (address->sa_family == AF_INET)
+        return IPAddress { reinterpret_cast<const struct sockaddr_in&>(address).sin_addr };
+    if (address->sa_family == AF_INET6)
+        return IPAddress { reinterpret_cast<const struct sockaddr_in6&>(address).sin6_addr };
```
```suggestion
        return IPAddress { reinterpret_cast<const struct sockaddr_in6*>(address)->sin6_addr };
```

achristensen07 committed:
Remove duplicate DNS lookup implementation
https://bugs.webkit.org/show_bug.cgi?id=257303
rdar://109812699

Reviewed by Youenn Fablet.

DNSServiceGetAddrInfo is the replacement for CFHostStartInfoResolution.

Remove NetworkRTCResolver in favor of implementing DNSResolveQueueCFNet::resolve
like the libSoup ports and sharing 1 implementation instead of having duplicate code.

* Source/WebCore/platform/network/DNS.cpp:
(WebCore::resolveDNS):
* Source/WebCore/platform/network/DNS.h:
(WebCore::IPAddress::fromSockAddrIn6): Deleted.
* Source/WebCore/platform/network/DNSResolveQueue.h:
* Source/WebCore/platform/network/cf/DNSResolveQueueCFNet.cpp:
(WebCore::DNSResolveQueueCFNet::CompletionHandlerWrapper::create):
(WebCore::DNSResolveQueueCFNet::CompletionHandlerWrapper::complete):
(WebCore::DNSResolveQueueCFNet::CompletionHandlerWrapper::addIPAddress):
(WebCore::DNSResolveQueueCFNet::CompletionHandlerWrapper::CompletionHandlerWrapper):
(WebCore::extractIPAddress):
(WebCore::dnsLookupCallback):
(WebCore::DNSResolveQueueCFNet::performDNSLookup):
(WebCore::DNSResolveQueueCFNet::platformResolve):
(WebCore::DNSResolveQueueCFNet::resolve):
(WebCore::DNSResolveQueueCFNet::stopResolve):
(WebCore::clientCallback): Deleted.
* Source/WebCore/platform/network/cf/DNSResolveQueueCFNet.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.cpp:
(WebKit::NetworkRTCProvider::close):
(WebKit::NetworkRTCProvider::createResolver):
(WebKit::NetworkRTCProvider::stopResolver):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCResolver.cpp: Removed.
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCResolver.h: Removed.
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCResolverCocoa.cpp: Removed.
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCResolverCocoa.h: Removed.
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/264586@main
